src/dom.ts

```diff
@@ -126,6 +127,48 @@ export function containsClass(element: HTMLElement, className: string): boolean
 	return targetClass.indexOf(' ' + className + ' ') > -1;
 }
 
+/*
+ * Creates an Element
+ *
+ * @param tagName Type of element to create
+ * @param kwArgs An object containing attributes to add to the Element. If an "attributes"
+ * property is present, each member of this sub-object will be added to the Element via element.setAttribute
+ * @param children An array of either Nodes or strings, the latter of which will be converted to Nodes via dom.fromString
+ *
+ * @example
+ * var div = dom.create('div', {'data-index': 0, attributes: {'class': 'loaded'}});
+ *
+ * @example
+ * var div = dom.create('ul', null, [dom.create('li'), dom.create('li')]);
+ * 
+ * @examples
+ * var div = dom.create('ul', null, [dom.create('li'), dom.create('li')]);
+ */
+export function create(tagName: string, kwArgs?: dom.CreateArgs, children?: (Node|string)[]): HTMLElement {
+	var element = document.createElement(tagName);
+	if (kwArgs) {
+		for (let property in kwArgs) {
+			if (property === 'attributes') {
+				for (let attribute in kwArgs.attributes) {
+					element.setAttribute(attribute, kwArgs.attributes[attribute]);
+				}
+			}
+			else {
+				(<any> element)[property] = kwArgs[property];
+			}
+		}
+	}
+	if (children) {
+		for (let child of children) {
+			if (typeof child === 'string') {
+				child = fromString(<string> child);
```

Comment:
IMO this should use `document.createTextNode` directly.  It's not intended to place HTML, which this would allow it to do.